src/main/java/com/example/gradecalculator/service/UserService.java

```diff
@@ -130,4 +137,17 @@ public User getAuthenticatedUser(Authentication authentication) {
             throw new IllegalArgumentException("User not found");
         }
     }
+
+    @Transactional
+    public boolean deleteUserAndSubjects(Long userId) {
+        Optional<User> userOptional = userRepository.findById(userId);
+        if (!userOptional.isPresent()) {
+            System.out.println("User not found with ID: " + userId);
```

Comment:
should these really just print to the console or throw exceptions?

Reply:
these are just an ad for better understanding what just really happened while development

Reply:
alwasy default to use logger! when developing locally you can configure the logger to print to a console output (if it not already happens on default)
Usage of logger is more flexible while console prints are not really usefull outside of the development cycle (fine to have them, but should be removed before finishing the implementation)